modules/imgimport/src/csvreading.cpp

```diff
@@ -0,0 +1,178 @@
+#include "csvreading.h"
+#include <fstream>
+#include <string>
+#include "metadatalinkedlist.h"
+#include <vector>
+#include <iostream>
+#include "metadata.h"
+#include <boost/log/trivial.hpp>
+using namespace std;
+
+Metadatalinkedlist* readcsv(const char* filename){
+    Metadatalinkedlist* result=new Metadatalinkedlist;
+    result=0;
```

Comment:
This still leaks memory. Whenever you create an object using the new keyword you allocate memory on the heap, which stays allocated until you use delete to free it. By immediately assigning the pointer to something else you are losing the location of the allocated memory, making it now impossible to free (a leak).

My point before was that you can initialize result and poi as null and then the first time through the loop (i.e. when poi == null) make both result and poi point to the temp node, the new head and tail of the 1 element list. On subsequent iterations you then just have to set poi->next = tmp and poi = tmp, adding the node to the list and updating poi to point at the new tail.
Then at the end you can just return result (the pointer to the head of the list).